frontend/src/containers/map/sidebar/analysis/analysis-buttons/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { useCallback } from 'react';
+
+import { useSetAtom } from 'jotai';
+import { useResetAtom } from 'jotai/utils';
+
+import { Button } from '@/components/ui/button';
+import { analysisAtom, drawStateAtom } from '@/containers/map/store';
+
+const AnalysisButtons: React.FC = () => {
+  const setDrawState = useSetAtom(drawStateAtom);
+  const resetAnalysis = useResetAtom(analysisAtom);
+
+  const resetDrawing = useCallback(() => {
+    setDrawState((prevState) => ({
+      ...prevState,
+      feature: null,
+    }));
+  }, [setDrawState]);
```

Comment:
Why are we resetting the atom here manually (and partially) instead of using the `useResetAtom` utility?

Reply:
The flow has changes a bit as per the meet on Tuesday (so these changes are now outdated), and it was uncertain at the time of mock implementation. 

But the idea here was that, when the user clicks _"Re-Draw"_, we would need to clear the feature but still keep the drawing mode active so that the user keeps drawing. 

In the current flow/implementation though, when we reset we now reset the drawing atom, as well as the analysis one, but we still need to keep analysis active. A new property was added to the analysis atom (`active`) for this. Still, the new implementation will first reset the atom and then set `active` back to true; although it works I don't think this is correct, needs another pair of eyes on it if you have the time.